### PR TITLE
Remove CRC in all possible places

### PR DESCRIPTION
**Pull request by heinrich5991**
Some places remain due to net/file format compatibility.

### CODE REVIEW

**Comment by heinrich5991** [comment]
Note: This will require a change on the maps server as well. It'll also have to offer files in the mapname_sha256.map format. This unifies the map filename format with vanilla.

**Comment by def-** [comment]
> This will require a change on the maps server as well.

Done.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1635,6 +1633,8 @@ if(CLIENT)
     demoedit.h
     friends.cpp
     friends.h
+    ghost.cpp
```
Comment (def-) [inline CMakeLists.txt:1636]: Will loading old ghost files still work?

File: src/engine/client/ghost.cpp [hunk 2]
```diff
@@ -219,14 +216,46 @@ int CGhostLoader::Load(const char *pFilename, const char *pMap, unsigned Crc)
...
+			sha256_str(MapSha256, aMapSha256, sizeof(aMapSha256));
+			char aBuf[256];
+			str_format(aBuf, sizeof(aBuf), "ghost map '%s' sha256 mismatch, wanted=%s ghost=%s", pMap, aMapSha256, aGhostSha256);
+			m_pConsole->Print(IConsole::OUTPUT_LEVEL_STANDARD, "ghost_loader", aBuf);
+			io_close(m_File);
+			m_File = 0;
+			return -1;
+		}
+	}
+	else
```
Comment (def-) [inline src/engine/client/ghost.cpp:244]: ok, that answers my question.

**Comment by def-** [comment]
Fine from my side. Anyone else want to take a look?

**Comment by heinrich5991** [comment]
Support for version 2 & 3 was removed, but all version 2 & 3 demos should be converted on every map load.

**Comment by Jupeyy** [comment]
[Aip-Gores_nameless tee_109.900_25771dbe.gho.zip](https://github.com/ddnet/ddnet/files/5376749/Aip-Gores_nameless.tee_109.900_25771dbe.gho.zip)

My ghost doesn't work, what client version should convert them?, i just tried the master branch, where it still worked

**Comment by def-** [comment]
@heinrich5991 ^

**Comment by heinrich5991** [comment]
This is a version 5 ghost which should work unmodified in the new client. I will investigate.

**Comment by heinrich5991** [comment]
@Jupeyy Should be fixed now, I forgot to rewind the file.

**Comment by Jupeyy** [comment]
Works, rest looks good i think

**Comment by def-** [comment]
bors r+
Thanks!

**Pull request merged**